warp10/src/main/java/io/warp10/script/functions/FIND.java

```diff
@@ -155,7 +155,8 @@ public Object apply(WarpScriptStack stack) throws WarpScriptException {
       if (this.metaset) {
         throw new WarpScriptException(getName() + " expects a list of parameters on top of the stack.");
       }
-      mapparams = true;
+      mapparams = ((Map)top).containsKey(FETCH.PARAM_TOKEN);
```

Comment:
What happens if the object in 'top' is not a Map?

Reply:
The check is done three lines upper, the cast is safe.
The functional problem is someone using this unsupported syntax with a label called "token" as a label selector. 

Unit tests : 
`[ $READTOKEN '~.*speed.*' { 'id' 'xaaacgz' } ] FIND` //supported syntax with list input: OK
`$READTOKEN '~.*speed.*' { 'id' 'xaaacgz' } FIND` //unsupported syntax: OK with this PR
`$READTOKEN '~.*speed.*' { 'token' 'xx' } FIND` //unsupported syntax, but will be seen as the new map input syntax. Will fail with the message  `missing parameters 'class', 'labels', 'selector' or 'selectors'`

Reply:
The check should be more specific to reduce the possible errors. Beyond checking 'token', check for either 'class'/'labels' or 'selector'/'selectors'